app/src/validation/user/userValidation.js

```diff
@@ -0,0 +1,52 @@
+"use strict";
+
+const { body, param } = require("express-validator");
+const { createValidation } = require("@middleware/validationHelper.js");
+
+const checkNicknameUpdate = createValidation(
+  body("newNickname").custom((value) => {
+    const korean = value.match(/[가-힣]/g) || [];
+    const others = value.replace(/[가-힣]/g, "");
+    if (korean.length > 10) {
+      throw new Error("닉네임 한글은 최대 10자까지 허용됩니다.");
+    }
+    if (others.length > 30) {
+      throw new Error("닉네임 영문/숫자는 최대 30자까지 허용됩니다.");
+    }
+    if (!/^[가-힣a-zA-Z0-9]+$/.test(value)) {
+      throw new Error("닉네임은 한글, 영어, 숫자만 허용됩니다.");
+    }
+    return true;
+  })
+);
+
+const checkPasswordUpdate = createValidation(
+  body("currentPassword")
+    .notEmpty()
+    .withMessage("현재 비밀번호를 입력해주세요.")
+    .isLength({ min: 8, max: 20 })
+    .withMessage("현재 비밀번호는 8 ~ 20 글자여야 합니다.")
+    .matches(/^(?=.*[A-Za-z])(?=.*\d)(?=.*[!@#$%^&*()\-_=+{}\[\]:;"'<>,.?/\\|])/)
+    .withMessage("현재 비밀번호는 영문, 숫자, 특수문자를 각각 1자 이상 포함해야 합니다."),
```

Comment:
현재 비밀번호는 이미 생성 시 유효성 검사를 통과한 값이기 때문에, 업데이트 시에는 별도의 검사 없이 공백 여부만 확인해주면 좋을 것 같습니다.
```suggestion
const checkPasswordUpdate = createValidation(
  body("currentPassword")
    .notEmpty()
    .withMessage("현재 비밀번호를 입력해주세요.")
```

Reply:
피드백 감사합니다!
말씀해주신 것처럼, 현재 비밀번호는 이미 생성 시 유효성 검사를 통과한 값이므로 업데이트 시에는 공백 여부만 확인해도 충분하다고 생각합니다.
제안해주신 방향으로 코드에 반영하겠습니다 👍